portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,14 +33,93 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("UserData");
+    ArrayList<UserData> data = new ArrayList<UserData>();	
+    PreparedQuery results = datastore.prepare(query);
+
+    Date currentDate = new Date();  
+    
+    for (Entity entity : results.asIterable()) { 
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
+      String email = (String) entity.getProperty("email");
+      String comment = (String) entity.getProperty("comment");
+      Date date = (Date) entity.getProperty("date");
+      UserData newData = new UserData(id, name, email, comment, date);
+      data.add(newData);
+    }
+
+    Gson gson = new Gson();
+    int numComments = filterCommentsByNum(request, "numValue", 4);
+    
     response.setContentType("application/json;");
-    // response.getWriter().println("Hello Chisom!");
-    String jsonComments = "{\"Femi\": \"Nice page Chisom\", \"Dera\": \"Try to make the web page reponsive\", \"Kamsi\": \"Kudos\"}";
-    response.getWriter().println(jsonComments);
+    if (numComments > data.size()){ 
```

Comment:
the code in both branches of this if statement is almost identical. is there a way you can change the logic so that the if statement isn't necessary?

generally when writing code we want to avoid unnecessary branching - branching is harder code to read and understand because now your reader has to keep more possibilities in their head, one for each branch of the if statement. many times branching is good of course - but if it's possible to trivially eliminate branches then that is generally preferred.

Reply:
I don't understand what you mean here. Can you throw some more light on it?

Reply:
done